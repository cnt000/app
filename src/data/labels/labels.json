{
  "openMenuLabel": "apri il menu",
  "title": "Pungilandia",
  "titleDescription": "Piante grasse, strane ed insolite",
  "goToCartLabel": "vai al carrello",
  "copyright": "Copyright © 2020",
  "privacy": "privacy",
  "contact": "contatti",
  "company": "Azienda Agricola Pungilandia Rocchi Domenico",
  "companyAddress": "Via Isonzo 11, 65123 Pescara P.IVA 01203140684 CF RCCDNC59R30I712B\nIscrizione REA Pescara Nr.97422",
  "disclaimer": "Azienda in possesso Registro di Detenzione di Specie Vegetali Incluse\nNegli Allegati CITES A e B del Regolamento (CE) 338/97 del Consiglio del 9\nDicembre 1996 e Successive Modifiche e Attuazioni. Registro N.000001/2001\nVAB",
  "mailToLink": "mailto:info@pungilandia.com",
  "privacyLink": "http://www.pungilandia.com",
  "addToCart": "Aggiungi al carrello",
  "searchPlants": "Cerca piante",
  "searchPlantsImage": "http://www.pungilandia.com/php/images/lo/cotyl_mediop.jpg",
  "hintsFromPungilandiaTitle": "I Consigli di Pungilandia",
  "articleBy": "A cura del dott. Mimmo Rocchi",
  "articleContent": "consigliColtivazione.md",
  "mimmoImage": "http://www.pungilandia.com/media/MIMMO.JPG",
  "menuChiSiamo": "Chi siamo",
  "menuListaPiante": "Lista piante",
  "menuNovita": "Novità",
  "menuProdottiSpeciali": "Prodotti speciali",
  "menuConsigliPerColtivare": "Consigli per coltivare",
  "menuOrdini": "Ordini",
  "menuFotoShow": "Foto show",
  "menuEmail": "E-mail"
}
