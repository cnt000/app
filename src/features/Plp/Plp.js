import React, { Suspense } from 'react';
import PropTypes from 'prop-types';

import useIntersect from '../../hook/useIntersect';
import Product from '../Product/Product';
import EmptyResults from '../EmptyResults/EmptyResults';
import CercaPiante from '../CercaPiante/CercaPiante';
import MyErrorBoundary from '../ErrorBoundary/ErrorBoundary';
import withFetch from './withFetch';
import Pagination from '../Pagination/Pagination';
import styles from './Plp.module.css';

const Plp = ({ products }) => {
  const [ref, entry] = useIntersect({
    threshold: 0.1,
  });

  const hasResults =
    !products.error && !products.code && products.message !== 'not found';

  const MemoizedPagination = React.memo(
    () =>
      entry.intersectionRatio > 0.1 && (
        <MyErrorBoundary>
          <Suspense
            fallback={<div className={styles.paginationBox}>Loading...</div>}
          >
            <Pagination />
          </Suspense>
        </MyErrorBoundary>
      ),
  );

  return (
    <main className={styles.plpContent} role="main">
      <div className={styles.pageHeader}>
        <div className={styles.left}>
          <h2 className={styles.title}>Le nostre piante</h2>
        </div>
        <CercaPiante />
      </div>
      <div className={styles.products}>
        {!hasResults && <EmptyResults />}
        {hasResults && products.map((product, i) => Product({ ...product, i }))}
      </div>
      <div
        ref={ref}
        ratio={entry.intersectionRatio}
        className={styles.paginationBox}
      >
        <MemoizedPagination />
      </div>
    </main>
  );
};

Plp.propTypes = {
  products: PropTypes.array,
  Link: PropTypes.object,
};

export default withFetch(Plp);
