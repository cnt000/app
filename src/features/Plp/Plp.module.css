.plpContent {
  grid-area: content;
  padding: 0.4rem;
}

.products {
  display: flex;
  flex-wrap: wrap;
}

.product {
  height: auto;
  width: 25%;
  text-align: center;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
}

.product .name {
  margin: 0.6rem auto;
  width: 300px;
  line-height: 1.2;
}

.addToCart {
  margin: 0 auto;
  margin: 1rem auto 2rem auto;
  width: 50%;
  height: 2rem;
}

.pageHeader {
  display: flex;
  justify-content: space-between;
  height: 3rem;
  margin: 1rem auto;
}

.pageHeader .title {
  font-family: 'Caveat Brush', sans-serif;
  font-size: 32px;
  color: var(--green);
}

.product .image {
  width: 50%;
}

.pageHeader .cercapiante span {
  margin-left: -2.5rem;
  position: relative;
  text-shadow: 1px 1px 2px var(--white), 0 0 0.2em var(--grey);
}

@media only screen and (min-width: 1025px) {
  .plp-content {
    grid-area: content;
    padding: 2rem;
  }

  .pageHeader .left {
    text-align: left;
  }

  .pageHeader {
    display: flex;
    justify-content: space-between;
    height: 5rem;
    margin: 0.6rem auto;
  }
}
