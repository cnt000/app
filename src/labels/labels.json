{
  "openMenuLabel": "apri il menu",
  "title": "Pungilandia",
  "titleDescription": "Piante grasse, strane ed insolite",
  "goToCartLabel": "vai al carrello",
  "copyright": "Copyright Â© 2020",
  "privacy": "privacy",
  "contact": "contatti",
  "company": "Azienda Agricola Pungilandia Rocchi Domenico",
  "companyAddress": "Via Isonzo 11, 65123 Pescara P.IVA 01203140684 CF RCCDNC59R30I712B<br>Iscrizione REA Pescara Nr.97422",
  "disclaimer": "Azienda in possesso Registro di Detenzione di Specie Vegetali Incluse<br>Negli Allegati CITES A e B del Regolamento (CE) 338/97 del Consiglio del 9<br>Dicembre 1996 e Successive Modifiche e Attuazioni. Registro N.000001/2001<br>VAB"
}
