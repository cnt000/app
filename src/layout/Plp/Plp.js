import React, { useContext } from 'react';
import PropTypes from 'prop-types';

import LabelsContext from '../../features/labels';
import styles from './Plp.module.css';

const Product = ({ i, productName, productImage, price }) => {
  const { addToCart } = useContext(LabelsContext);
  return (
    <article key={`${productName}_${i}`} className={styles.product}>
      <a href="/item" className={styles.imageContainer}>
        <img className={styles.image} src={productImage} alt={productName} />
        <h1 className={styles.name}>{productName}</h1>
        <div>{price}</div>
      </a>
      <button className={styles.addToCart}>{addToCart}</button>
    </article>
  );
};

Product.propTypes = {
  i: PropTypes.number,
  productName: PropTypes.string,
  productImage: PropTypes.string,
  price: PropTypes.string,
  addToCart: PropTypes.string,
};

const Plp = ({ productName, productImage, price }) => {
  const { searchPlants, searchPlantsImage } = useContext(LabelsContext);
  return (
    <main className={styles.plpContent} role="main">
      <div className={styles.pageHeader}>
        <div className={styles.left}>
          <h2 className={styles.title}>Le nostre piante</h2>
          <span className={styles.lastmodified}>02/04/2020</span>
        </div>
        <div className={styles.cercapiante}>
          <a href="/search" title={searchPlants}>
            <img src={searchPlantsImage} alt={searchPlants} />
            <span>{searchPlants}</span>
          </a>
        </div>
      </div>
      <div className={styles.products}>
        {Array(12)
          .fill(' ')
          .map((_, i) => Product({ i, productName, productImage, price }))}
      </div>
    </main>
  );
};

Plp.propTypes = {
  productName: PropTypes.string,
  productImage: PropTypes.string,
  price: PropTypes.string,
  addToCart: PropTypes.string,
  searchPlants: PropTypes.string,
  searchPlantsImage: PropTypes.string,
};

export default Plp;
