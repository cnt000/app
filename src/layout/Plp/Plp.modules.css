.page__product-info {
  text-align: center;
}

.page__products {
  display: flex;
  flex-wrap: wrap;
}

.product {
  width: 50%;
  text-align: center;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
}

.product__name {
  margin: 0.6rem auto;
  width: 80%;
  line-height: 1.2;
}

.product__addtocart {
  margin: 0 auto;
  margin: 1rem auto 2rem auto;
  width: 50%;
  height: 2rem;
}

.page-header {
  display: flex;
  justify-content: space-between;
  height: 3rem;
  margin: 1rem auto;
}

.page-header__title {
  font-family: 'Caveat Brush', sans-serif;
  font-size: 32px;
  color: var(--green);
}

.product .image {
  width: 90%;
}

@media only screen and (min-width: 1025px) {
  .plp-content {
    grid-area: content;
    padding: 2rem;
  }

  .page-header__left {
    text-align: left;
  }
}
