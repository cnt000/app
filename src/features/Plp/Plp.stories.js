import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';

import Plp from './Plp';
import labels from '../../data/labels/labels.json';
import products from '../../data/products/searchpage.json';
import LabelsContext from '../Labels';

export default {
  component: Plp,
  title: 'Plp',
};

export const Default = () => (
  <LabelsContext.Provider value={labels}>
    <Router>
      <Plp products={products} />
    </Router>
  </LabelsContext.Provider>
);
