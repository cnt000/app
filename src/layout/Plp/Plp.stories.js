import React from 'react';

import Plp from './Plp';
import labels from '../../labels/labels.json';

const productInfo = {
  productName: 'Adromischus clavifolius',
  productImage:
    'http://www.pungilandia.com/php/images/hi/adromischus-clavifolius.jpg',
  vaseDiameter: 'Vaso 24cm',
  price: 'Prezzo: â‚¬ 9,00',
};

export default {
  component: Plp,
  title: 'Plp',
};

export const Default = () => <Plp {...labels} {...productInfo} />;
