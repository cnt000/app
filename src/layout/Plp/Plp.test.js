import React from 'react';
import renderer from 'react-test-renderer';

import Plp from './Plp';
import LabelsContext from '../../features/labels';
import labels from '../../data/labels/labels.json';
import product from '../../data/products/clavifolius.json';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <LabelsContext.Provider value={labels}>
        <Plp {...product} />
      </LabelsContext.Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

