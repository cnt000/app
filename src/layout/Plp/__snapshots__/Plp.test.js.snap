// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<main
  className="plpContent"
  role="main"
>
  <div
    className="pageHeader"
  >
    <div
      className="left"
    >
      <h2
        className="left"
      >
        Le nostre piante
      </h2>
      <span
        className="lastmodified"
      >
        02/04/2020
      </span>
    </div>
    <div
      className="cercapiante"
    >
      <a
        href="/search"
        title="cerca piante"
      >
        <img
          alt="cerca piante"
          src="http://www.pungilandia.com/php/images/lo/cotyl_mediop.jpg"
        />
        <span>
          il cerca piante
        </span>
      </a>
    </div>
  </div>
  <div
    className="products"
  >
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/item"
      >
        <img
          alt="Adromischus  clavifolius"
          className="image"
          src="http://www.pungilandia.com/php/images/hi/adromischus-clavifolius.jpg"
        />
        <h1
          className="name"
        >
          Adromischus clavifolius Ų 24
        </h1>
        <div>
          € 9,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/item"
      >
        <img
          alt="Adromischus  clavifolius"
          className="image"
          src="http://www.pungilandia.com/php/images/hi/adromischus-clavifolius.jpg"
        />
        <h1
          className="name"
        >
          Adromischus clavifolius Ų 24
        </h1>
        <div>
          € 9,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/item"
      >
        <img
          alt="Adromischus  clavifolius"
          className="image"
          src="http://www.pungilandia.com/php/images/hi/adromischus-clavifolius.jpg"
        />
        <h1
          className="name"
        >
          Adromischus clavifolius Ų 24
        </h1>
        <div>
          € 9,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/item"
      >
        <img
          alt="Adromischus  clavifolius"
          className="image"
          src="http://www.pungilandia.com/php/images/hi/adromischus-clavifolius.jpg"
        />
        <h1
          className="name"
        >
          Adromischus clavifolius Ų 24
        </h1>
        <div>
          € 9,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/item"
      >
        <img
          alt="Adromischus  clavifolius"
          className="image"
          src="http://www.pungilandia.com/php/images/hi/adromischus-clavifolius.jpg"
        />
        <h1
          className="name"
        >
          Adromischus clavifolius Ų 24
        </h1>
        <div>
          € 9,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/item"
      >
        <img
          alt="Adromischus  clavifolius"
          className="image"
          src="http://www.pungilandia.com/php/images/hi/adromischus-clavifolius.jpg"
        />
        <h1
          className="name"
        >
          Adromischus clavifolius Ų 24
        </h1>
        <div>
          € 9,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/item"
      >
        <img
          alt="Adromischus  clavifolius"
          className="image"
          src="http://www.pungilandia.com/php/images/hi/adromischus-clavifolius.jpg"
        />
        <h1
          className="name"
        >
          Adromischus clavifolius Ų 24
        </h1>
        <div>
          € 9,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/item"
      >
        <img
          alt="Adromischus  clavifolius"
          className="image"
          src="http://www.pungilandia.com/php/images/hi/adromischus-clavifolius.jpg"
        />
        <h1
          className="name"
        >
          Adromischus clavifolius Ų 24
        </h1>
        <div>
          € 9,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/item"
      >
        <img
          alt="Adromischus  clavifolius"
          className="image"
          src="http://www.pungilandia.com/php/images/hi/adromischus-clavifolius.jpg"
        />
        <h1
          className="name"
        >
          Adromischus clavifolius Ų 24
        </h1>
        <div>
          € 9,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/item"
      >
        <img
          alt="Adromischus  clavifolius"
          className="image"
          src="http://www.pungilandia.com/php/images/hi/adromischus-clavifolius.jpg"
        />
        <h1
          className="name"
        >
          Adromischus clavifolius Ų 24
        </h1>
        <div>
          € 9,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/item"
      >
        <img
          alt="Adromischus  clavifolius"
          className="image"
          src="http://www.pungilandia.com/php/images/hi/adromischus-clavifolius.jpg"
        />
        <h1
          className="name"
        >
          Adromischus clavifolius Ų 24
        </h1>
        <div>
          € 9,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
    <article
      className="product"
    >
      <a
        className="imageContainer"
        href="/item"
      >
        <img
          alt="Adromischus  clavifolius"
          className="image"
          src="http://www.pungilandia.com/php/images/hi/adromischus-clavifolius.jpg"
        />
        <h1
          className="name"
        >
          Adromischus clavifolius Ų 24
        </h1>
        <div>
          € 9,00
        </div>
      </a>
      <button
        className="addToCart"
      >
        Aggiungi al carrello
      </button>
    </article>
  </div>
</main>
`;
